semantic_conventions/http-common.yaml

```
@@ -5,10 +5,51 @@ groups:
     prefix: http
     attributes:
       - id: request.method
-        type: string
+        type:
+          allow_custom_values: true
+          members:
+            - id: connect
+              value: "CONNECT"
+              brief: 'CONNECT method.'
+            - id: delete
+              value: "DELETE"
+              brief: 'DELETE method.'
+            - id: get
+              value: "GET"
+              brief: 'GET method.'
+            - id: head
+              value: "HEAD"
+              brief: 'HEAD method.'
+            - id: options
+              value: "OPTIONS"
+              brief: 'OPTIONS method.'
+            - id: patch
+              value: "PATCH"
+              brief: 'PATCH method.'
+            - id: post
+              value: "POST"
+              brief: 'POST method.'
+            - id: put
+              value: "PUT"
+              brief: 'PUT method.'
+            - id: trace
+              value: "TRACE"
+              brief: 'TRACE method.'
+            - id: other
+              value: "other"
+              brief: 'Any custom HTTP method that the instrumentation has no prior knowledge of.'
         requirement_level: required
         brief: 'HTTP request method.'
         examples: ["GET", "POST", "HEAD"]
+        note: |
+          HTTP request method SHOULD be one of the methods defined in [RFC9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-methods)
+          or the PATCH method defined in [RFC5789](https://www.rfc-editor.org/rfc/rfc5789.html).
+          Instrumentation MAY additionally support the closed set of custom HTTP methods defined in
+          [HTTP method registry](https://www.iana.org/assignments/http-methods/http-methods.xhtml) or a private registry.
+          If the HTTP request method is not known to the instrumentation, it MUST set the `http.request.method` attribute to `other` and SHOULD
```

```suggestion
          If the HTTP request method is not known to the instrumentation, it SHOULD set the `http.request.method` attribute to `other` and SHOULD
```

The discussion on this PR is moving a bit off scope of these specific semconv changes, but I generally think this lgtm and is in line with @reyang recommendations here for how the semconv should be changed. however, If i'm understanding their feedback correctly, the goal would be not to enforce this as a requirement with `MUST`, but make it an allowed option for language sigs to implement.

---

Whenever the method is modified in any way (changing to "other" or maybe case correction) the request.method MUST be set, and this should not be optional as it is crucial to know the original method, as it is interesting _especially_ in nonstandard cases. See also https://github.com/open-telemetry/semantic-conventions/pull/17#issuecomment-1570821097

---

```suggestion
          If the HTTP request method is not known to the instrumentation, it MUST set the `http.request.method` attribute to `other` and MUST
```

This is IMHO a very important requirement, even fundamental to this "Option A".